robotintelligence.cpp

```diff
@@ -20,15 +21,22 @@ RobotIntelligence::~RobotIntelligence() {
 
 void RobotIntelligence::run() {
 	initParticles();
+    double lastTime = timeData.getTime();
+    unsigned int lastMeasurementId = laserData.getMeasurementId()-1;
 	while(true) { //TODO
-		std::vector<double> sensorData = readSensors();
-		evalSensors(sensorData);
+        if(lastMeasurementId == laserData.getMeasurementId())
+            continue;
+
+        lastMeasurementId = laserData.getMeasurementId();

```

Comment:
in future not relevant, but ID must be received at the same time at the laserData, any other use makes no sense